api/hypershift/v1beta1/openstack.go

```diff
@@ -136,6 +140,17 @@ type OpenStackPlatformSpec struct {
 	// +kubebuilder:validation:XValidation:rule="isIP(self)",message="floatingIP must be a valid IPv4 or IPv6 address"
 	// +optional
 	IngressFloatingIP string `json:"ingressFloatingIP,omitempty"`
+
+	// ImageRetentionPolicy defines the policy for handling resources associated with the image
+	// when the cluster is deleted.
+	// The default (set by the Nodepool controller) matches OpenStack Resource Controller (ORC)
+	// behavior where the OpenStack Glance Image is pruned when the Image object is deleted during the
+	// HostedCluster deletion.
+	// It is defined at the HostedCluster level and will be used for all nodepools images
```

Comment:
can we explicitly document what happens if this is changed day 2?

Reply:
Yes, and I also confirmed that the retention policy is mutable in ORC:
https://github.com/k-orc/openstack-resource-controller/blob/854b83db2bf6622d58f8583a643bdc1cab57b3fc/cmd/resource-generator/data/api.template#L78-L80